lib/size_checker.rb

```diff
@@ -19,38 +24,42 @@ def check(options)
     private
 
     def calculate_local_size(dist_folder, excludes)
+      puts get_entries(dist_folder)
       get_entries(dist_folder).reject { |entry| excludes.include? entry }
-                              .map { |entry| File.directory?(entry) ? calculate_dir_size(entry) : File.size(entry) }
+                              .map { |entry| File.directory?(dist_folder + File::SEPARATOR + entry) ?
+                                               (
+                                                 puts "calculate_dir_size #{(dist_folder + File::SEPARATOR + entry)}: #{calculate_dir_size(dist_folder + File::SEPARATOR + entry)}"
+                                                 calculate_dir_size(dist_folder + File::SEPARATOR + entry)
+                                               ) :
+                                               File.size(dist_folder + File::SEPARATOR + entry) }
                               .sum
     end
 
     def calculate_dir_size(dir)
-      size = File.size(dir)
+      size = 0
       get_entries(dir).each do |entry|
-        path = "#{dir}/#{entry}"
+        path = "#{dir}#{File::SEPARATOR}#{entry}"
+
         size += if File.directory? path
                   calculate_dir_size(path)
                 else
                   File.size(path)
                 end
       end
-
       size
     end
 
     def get_entries(dir)
       Dir.entries(dir).select { |value| value != '.' && value != '..' }
     end
 
-    def calculate_remote_excludes_size(user, host, dirs)
-      return 0 if dirs.empty?
+    def calculate_remote_excludes_size(user, host, excludes)
+      return 0 if excludes.empty?
+
+      exclude_options = (excludes).map { |exclude| "--exclude=#{exclude}" }.join(' ')
 
       Net::SSH.start(host, user[:username], password: user[:password], verify_host_key: :never) do |ssh|
-        ssh.exec!('shopt -s dotglob; du -sb *')
-           .scan(/(\d+)\s+(\S+)/)
-           .select { |_, path| dirs.include? path }
-           .map { |size, _| size.to_i }
-           .sum
+        ssh.exec!("shopt -s dotglob; expr $(du -sb . | cut -f1 ) - $(du -sb . #{exclude_options} | cut -f1)").to_i
```

Comment:
The `shopt -s dotglob` is obsolete with this change. This changes the behavior of the shell regarding an asterisk (`*`).